### PR TITLE
feat(layout): add support for container and icon tokens

### LINKED ISSUE #4599: Add support for container and icon tokens in layout
This issue is for introducing experimental design tokens for container and icon sizes in the design system, specifically under the layout package. Once added, we'll integrate layout into our themes package so that spacing, layout, container, and icon tokens are all shipped through our theme.

Specific tokens and their values:

```scss
$container-01: rem(24px);
$container-02: rem(32px);
$container-03: rem(40px);
$container-04: rem(48px);
$container-05: rem(64px);

$icon-size-01: rem(16px);
$icon-size-02: rem(20px);
```

### PR DESCRIPTION
**Pull request by joshblack**
Closes #4599

Add in support for two experimental design tokens: `container-*` and `icon-size-*`. This also refactors our `layout` package to generate token values and adds in tests for our public exports.

#### Changelog

**New**

- `layout`
  - Add tests for token public exports

**Changed**

- `layout`
  - All spacing tokens are generated from our JavaScript
- `scss-generator`
  - Update build logic for identifiers in a Sass list

**Removed**

### CODE REVIEW

**Comment by joshblack** [comment]
@jnm2377 let me know if you want to walk through this PR or not! I know some of the theming stuff is random 😅

File: packages/layout/__tests__/layout-test.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { createSassRenderer, convert } from '@carbon/test-utils/scss';
+import { spacing, layout, fluidSpacing, container, iconSize } from '../src';
+
+function formatStep(name, index) {
+  let step = '' + index;
```
Comment (emyarod) [inline packages/layout/__tests__/layout-test.js]:
```suggestion
  let step = index;
```

couldn't we just set this as `index`? then the comparison on the next line would be between two numbers and then either L14 or L16 would turn `step` into a string anyway

**Review by emyarod — COMMENTED** [review]
do we need to run other build scripts to regenerate the files for the components vendor folders?

File: packages/layout/tasks/build.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+ * Format the given step for a token name. Most often, this is to pad a `0` for
+ * numbers that are less than 10. For example, instead of spacing-1 we would
+ * want spacing-01
+ *
+ * @param {string} name
+ * @param {number} index
+ * @returns {string}
+ */
+function formatStep(name, index) {
+  let step = '' + index;
```
Comment (joshblack) [inline packages/layout/tasks/build.js]:
```suggestion
  let step = index;
```

**Comment by joshblack** [comment]
@emyarod fair, let me do a full rebuild

**Comment by joshblack** [comment]
@emyarod just pushed

**Comment by joshblack** [comment]
Just a quick note for reviewers, only files that are important:

https://github.com/carbon-design-system/carbon/pull/4608/files#diff-6a437ff649f7fb0090d1d19912e9b7b0

https://github.com/carbon-design-system/carbon/pull/4608/files#diff-5d01ca9dc8b49bec4651ed6272ec98db

https://github.com/carbon-design-system/carbon/pull/4608/files#diff-14706e584d2f507f9020ee454cb12739

https://github.com/carbon-design-system/carbon/pull/4608/files#diff-3d381ead555374c7b314d410a0556237

https://github.com/carbon-design-system/carbon/pull/4608/files#diff-a868f9c4ca65d2ac76940dd81722f6af

https://github.com/carbon-design-system/carbon/pull/4608/files#diff-e28b5638ec7fd346acc156d1eb578eb5

**Review by emyarod — APPROVED** [review]
looks good to me, thanks for regenerating the `vendor` files

**Review by abbeyhrt — APPROVED** [review]
To the best of my knowledge, this looks good to me!

**Pull request merged**